specification/loadtestservice/LoadTestService.Management/models.tsp

```diff
@@ -0,0 +1,311 @@
+
+  @doc("The endpoints for this service to which the Batch service makes outbound calls.")
+  @visibility("read")
+  @extension("x-ms-identifiers", ["domainName"])
+  endpoints?: EndpointDependency[];
+}
+
+@doc("All identity configuration for Customer-managed key settings defining which identity should be used to auth to Key Vault.")
+model EncryptionPropertiesIdentity {
+  @doc("Managed identity type to use for accessing encryption key Url.")
+  type?: type;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "We need resourceId to be nullable."
+  @doc("User assigned identity to use for accessing key encryption key Url. Ex: /subscriptions/fa5fc227-a624-475e-b696-cdd604c735bc/resourceGroups/<resource group>/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myId.")
+  resourceId?: string | null;
+}
+
+@doc("Key and identity details for Customer Managed Key encryption of load test resource.")
+model EncryptionProperties {
+  @doc("All identity configuration for Customer-managed key settings defining which identity should be used to auth to Key Vault.")
+  identity?: EncryptionPropertiesIdentity;
+
+  @doc("key encryption key Url, versioned. Ex: https://contosovault.vault.azure.net/keys/contosokek/562a4bb76b524a1493a6afe8e536ee78 or https://contosovault.vault.azure.net/keys/contosokek.")
+  keyUrl?: string;
+}
+
+@doc("Quota bucket resource properties.")
+model QuotaResourceProperties {
+  @minValue(0)
+  @doc("Current quota limit of the quota bucket.")
+  limit?: int32;
+
+  @doc("Current quota usage of the quota bucket.")
+  @minValue(0)
+  usage?: int32;
+
+  @doc("Resource provisioning state.")
+  @visibility("read")
+  provisioningState?: ResourceState;
+}
+
+@doc("The location of quota resource")
+@subscriptionResource
+model Location {
+  @doc("The location of quota in ARM Normalized format like eastus, southeastasia etc.")
+  @key("location")
+  @path
+  @segment("locations")
+  @visibility("query", "read", "update")
+  name: string;
+}
+
+@doc("Quota bucket details object.")
+@parentResource(Location)
+model QuotaResource is ProxyResource<QuotaResourceProperties> {
+  @doc("The quota name.")
+  @key("quotaBucketName")
+  @path
+  @segment("quotas")
+  @extension("x-ms-parameter-location", "method")
+  name: string;
+}
+
+@doc("LoadTest resource properties.")
+model LoadTestProperties {
+  @maxLength(512)
+  @doc("Description of the resource.")
+  description?: string;
+
+  @visibility("read")
+  @doc("Resource provisioning state.")
+  provisioningState?: ResourceState;
+
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "URI is an abbervation and should be upper case"
+  @visibility("read")
+  @maxLength(2083)
+  @doc("Resource data plane URI.")
+  dataPlaneURI?: string;
+
+  @doc("CMK Encryption property.")
+  encryption?: EncryptionProperties;
+}
+
+#suppress "@azure-tools/typespec-azure-core/composition-over-inheritance" "CheckQuotaAvailabilityResponse has a allOf relationship with Resource"
```

Comment:
I don't think it *is* supposed to inherit resource? Isn't CheckQuotaAvailabilityRespones the response to a POST request? So it wouldn't need to be a resource type.

Reply:
The response has properties of resource i.e. id, name, type and systemdata. We can spread in Resource instead of inheriting but thats how it is in our existing published swagger. If this won't cause a breaking change to SDKs we can definitely remove the inheritance.

Reply:
I have removed inheritance of remove instead spread in the ArmResource model.